api/src/paths/project/{projectId}/delete.ts

```diff
@@ -0,0 +1,143 @@
+import { getSurveyAttachmentS3Keys } from './survey/{surveyId}/delete';
+import { deleteProjectSQL } from '../../../queries/project/project-delete-queries';
+import { deleteFileFromS3 } from '../../../utils/file-utils';
+
+const defaultLog = getLogger('/api/project/{projectId}/delete');
+
+export const DELETE: Operation = [deleteProject()];
+
+DELETE.apiDoc = {
+  description: 'Delete a project.',
+  tags: ['project'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  parameters: [
+    {
+      in: 'path',
+      name: 'projectId',
+      schema: {
+        type: 'number'
+      },
+      required: true
+    }
+  ],
+  responses: {
+    200: {
+      description: 'Boolean true value representing successful deletion.',
+      content: {
+        'application/json': {
+          schema: {
+            title: 'Project delete response',
+            type: 'boolean'
+          }
+        }
+      }
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function deleteProject(): RequestHandler {
+  return async (req, res) => {
+    defaultLog.debug({ label: 'Delete project', message: 'params', req_params: req.params });
+
+    if (!req.params.projectId) {
+      throw new HTTP400('Missing required path param `projectId`');
+    }
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      /**
+       * PART 1
+       * Get the attachment S3 keys for all attachments associated to this project and surveys under this project
+       * Used to delete them from S3 separately later
+       */
+      const getProjectAttachmentSQLStatement = getProjectAttachmentsSQL(Number(req.params.projectId));
+      const getSurveyIdsSQLStatement = getSurveyIdsSQL(Number(req.params.projectId));
+
+      if (!getProjectAttachmentSQLStatement || !getSurveyIdsSQLStatement) {
+        throw new HTTP400('Failed to build SQL get statement');
+      }
+
+      await connection.open();
+
+      const getProjectAttachmentsResult = await connection.query(
+        getProjectAttachmentSQLStatement.text,
+        getProjectAttachmentSQLStatement.values
+      );
+
+      if (!getProjectAttachmentsResult || !getProjectAttachmentsResult.rows) {
+        throw new HTTP400('Failed to get project attachments');
+      }
+
+      const getSurveyIdsResult = await connection.query(getSurveyIdsSQLStatement.text, getSurveyIdsSQLStatement.values);
+
+      if (!getSurveyIdsResult || !getSurveyIdsResult.rows) {
+        throw new HTTP400('Failed to get survey ids associated to project');
+      }
+
+      const surveyAttachmentS3Keys: string[] = Array.prototype.concat.apply(
+        [],
+        await Promise.all(
+          getSurveyIdsResult.rows.map((survey: any) => getSurveyAttachmentS3Keys(survey.id, connection))
+        )
+      );
```

Comment:
Can this not be simplified to something like:

```typescript
 const surveyAttachmentS3Keys: string[] = await Promise.all(
    getSurveyIdsResult.rows.map((survey: any) => getSurveyAttachmentS3Keys(survey.id, connection))
  );
```

Reply:
it complains that they are incompatible types:
```
Type 'any[][]' is not assignable to type 'string[]'.
Type 'any[]' is not assignable to type 'string'.
```

Reply:
it returns nested array that I am just flattening